### ISSUE TITLE
ValueError when attempting to validate list of validator and validator

### ISSUE THREAD

**Issue by rHorsey** [23andMe/Yamale#63]
Hi all;

In an issue similar (I think) to #22 we're running into situations where we have schemas with optional lists - i.e. 
```yml
example1:
  logic:
    or:
      - thing1
      - thing2
example2:
  logic:
    - or:
      - thing3
      - thing4
      - thing5
```
validating against:
```yml
example1: include('spec-1', required=False)
example2: include('spec-2', required=False)
---
spec-1:
  logic: >
    any(
      include('or-spec', required=False),
      required=False
    )
spec-2:
  logic: >
    any(
      list(
        include('or-spec', required=False),
        required=False
      ),
      required=False
    )
blended-spec:
  logic: >
    any(
      list(
        include('or-spec', required=False),
        required=False
      ),
      include('or-spec', required=False),
      required=False
    )
or-spec:
  or: any(list(str(required=True)), str(required=False))
```

I can create validators for either `example1` or `example2` (`spec-1` and `spec-2` respectively) but when I try to combine those together (`blended-spec`) I end up with the below stack trace. I'm starting to convince myself that it's not me, but I'd say it's still 50/50 - my YAML comprehension is really truly bad. Any advice would be greatly appreciated!

```
Traceback (most recent call last):
  File "/Users/hhorsey/.pyenv/versions/idsm3/lib/python3.7/site-packages/yamale/util.py", line 81, in get_value
    return reduce(getitem, path, dic)[last_key]
TypeError: list indices must be integers or slices, not str

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/hhorsey/repos/tracked/buildstockbatch-dev/buildstockbatch/base.py", line 294, in validate_project_schema
    return yamale.validate(schema, data)
  File "/Users/hhorsey/.pyenv/versions/idsm3/lib/python3.7/site-packages/yamale/yamale.py", line 39, in validate
    schema.validate(d)
  File "/Users/hhorsey/.pyenv/versions/idsm3/lib/python3.7/site-packages/yamale/schema/schema.py", line 49, in validate
    errors += self._validate(validator, data, key=key, includes=self.includes)
  File "/Users/hhorsey/.pyenv/versions/idsm3/lib/python3.7/site-packages/yamale/schema/schema.py", line 83, in _validate
    return self._validate_item(validator, data_item, position, includes)
  File "/Users/hhorsey/.pyenv/versions/idsm3/lib/python3.7/site-packages/yamale/schema/schema.py", line 104, in _validate_item
    includes, position)
  File "/Users/hhorsey/.pyenv/versions/idsm3/lib/python3.7/site-packages/yamale/schema/schema.py", line 150, in _validate_include
    errors += include_schema._validate(validator, data, includes=includes, key=key, position=pos)
  File "/Users/hhorsey/.pyenv/versions/idsm3/lib/python3.7/site-packages/yamale/schema/schema.py", line 83, in _validate
    return self._validate_item(validator, data_item, position, includes)
  File "/Users/hhorsey/.pyenv/versions/idsm3/lib/python3.7/site-packages/yamale/schema/schema.py", line 112, in _validate_item
    includes, position)
  File "/Users/hhorsey/.pyenv/versions/idsm3/lib/python3.7/site-packages/yamale/schema/schema.py", line 163, in _validate_any
    err = self._validate_item(v, data, pos, includes)
  File "/Users/hhorsey/.pyenv/versions/idsm3/lib/python3.7/site-packages/yamale/schema/schema.py", line 104, in _validate_item
    includes, position)
  File "/Users/hhorsey/.pyenv/versions/idsm3/lib/python3.7/site-packages/yamale/schema/schema.py", line 150, in _validate_include
    errors += include_schema._validate(validator, data, includes=includes, key=key, position=pos)
  File "/Users/hhorsey/.pyenv/versions/idsm3/lib/python3.7/site-packages/yamale/schema/schema.py", line 75, in _validate
    data_item = util.get_value(data, key)
  File "/Users/hhorsey/.pyenv/versions/idsm3/lib/python3.7/site-packages/yamale/util.py", line 83, in get_value
    return reduce(getitem, path, dic)[int(last_key)]
ValueError: invalid literal for int() with base 10: 'or'
```

I'm testing against [1.10.0](https://github.com/23andMe/Yamale/releases/tag/1.10.0). Thanks so much!

Ry

**Comment by drmull** [comment]
I tested your example (changed example1 and example2 to use the blended-spec) with the code from PR #60 and it validates successfully without any crash.

**Comment by rHorsey** [comment]
@drmull Brilliant! Thanks so much. I should have caught the include map fix in #60 - would you prefer I close this now, or wait for @mildebrandt to merge #60 first?

**Comment by drmull** [comment]
@rHorsey I would keep it open until the PR is merged

**Comment by mildebrandt** [comment]
This is available in version 2.0.
